src/test/java/hudson/plugins/sonar/SonarInstallationTest.java

```diff
@@ -43,29 +50,37 @@
 public class SonarInstallationTest extends SonarTestCase {
 
   @Test
-  public void testRoundtrip() throws IOException {
+  public void testRoundtrip() throws IOException, ExecutionException, InterruptedException {
     TriggersConfig triggers = new TriggersConfig();
     SonarGlobalConfiguration d = new SonarGlobalConfiguration();
+    String credentialsId = "credentialsId";
     SonarInstallation inst = spy(new SonarInstallation(
       "Name",
       "server.url",
-      "credentialsId",
+            credentialsId,
       null,
       "secretId",
       "mojoVersion",
       "props",
       "key=value",
       triggers));
-    StringCredentials cred = new StringCredentialsImpl(CredentialsScope.GLOBAL, "an-id", null, Secret.fromString("token"));
-    doReturn(cred).when(inst).getCredentials(any(Run.class));
+    StringCredentials cred = new StringCredentialsImpl(CredentialsScope.GLOBAL, credentialsId, null, Secret.fromString("token"));
+
+    CredentialsStore store = CredentialsProvider.lookupStores(j.jenkins).iterator().next();
+    store.addCredentials(Domain.global(), cred);
+
+    FreeStyleProject freeStyleProject = j.createFreeStyleProject();
+    FreeStyleBuild run = freeStyleProject.scheduleBuild2(0).get();
+    j.waitForCompletion(run);
+
     d.setInstallations(inst);
 
     SonarInstallation i = new SonarGlobalConfiguration().getInstallations()[0];
     String storedConfig = Util.loadFile(new File(Jenkins.getInstance().getRootDir(), d.getId() + ".xml"));
 
     assertThat(i.getName()).isEqualTo("Name");
     assertThat(i.getServerUrl()).isEqualTo("server.url");
-    assertThat(i.getServerAuthenticationToken(mock(Run.class))).isEqualTo("token");
+    assertThat(i.getServerAuthenticationToken(run)).isEqualTo("token");
```

Comment:
this was returning null instead of the token because of some unmocked methods I assume. Given we are using JenkinsRule here best to avoid mocking since there's no need to mock at all.